src/main/groovy/com/github/spotbugs/snom/SpotBugsTask.groovy

```diff
@@ -416,4 +429,28 @@ class SpotBugsTask extends DefaultTask implements VerificationTask {
         }
         return new StringBuilder().append(Character.toLowerCase(prunedName.charAt(0))).append(prunedName.substring(1)).toString()
     }
+
+    private processFilterFile(Property<Object> filterProperty,
+            RegularFileProperty filterFileProperty, String filterName) {
+        if (filterProperty.isPresent()) {
+            def filter = filterProperty.get()
+
+            if (filter instanceof File) {
+                filterFileProperty.set(filter)
+            }
+            else if (filter instanceof URL) {
+                def tempFile = File.createTempFile("spotBugs-$filterName", ".xml")
+                tempFile.deleteOnExit()
+
+                tempFile.withOutputStream { out ->
+                    out << filter.openStream()
+                }
+
+                filterFileProperty.set(tempFile)
+            }
+            else {
+                throw new IllegalArgumentException("The '$filterName' property must be a File or URL.")
```

Comment:
This exception is thrown to Gradle core (our code does not catch this exception), then I suggest using a subclass of `GradleException` like [InvalidUserDataException](https://docs.gradle.org/current/javadoc/org/gradle/api/InvalidUserDataException.html).

Reply:
What difference will that make?